index.js

```diff
@@ -40,39 +42,57 @@ const meow = (helpText, options) => {
 		hardRejection();
 	}
 
-	const minimistFlags = options.flags && typeof options.booleanDefault !== 'undefined' ? Object.keys(options.flags).reduce(
-		(flags, flag) => {
-			if (flags[flag].type === 'boolean' && !Object.prototype.hasOwnProperty.call(flags[flag], 'default')) {
-				flags[flag].default = options.booleanDefault;
-			}
+	const minimistFlags = Object.entries(options.flags).reduce((flags, [flagKey, flagValue]) => {
+		const flag = {...flagValue};
+		const {booleanDefault} = options;
 
-			return flags;
-		},
-		options.flags
-	) : options.flags;
+		if (
+			typeof booleanDefault !== 'undefined' &&
+			flag.type === 'boolean' &&
+			!Object.prototype.hasOwnProperty.call(flag, 'default')
+		) {
+			flag.default = flag.multiple ? [booleanDefault] : booleanDefault;
+		}
+
+		if (flag.multiple) {
+			flag.type = 'array';
+			delete flag.multiple;
+		}
+
+		flags[flagKey] = flag;
 
-	let minimistoptions = {
+		return flags;
+	}, {});
+
+	let minimistOptions = {
 		arguments: options.input,
 		...minimistFlags
 	};
 
-	minimistoptions = decamelizeKeys(minimistoptions, '-', {exclude: ['stopEarly', '--']});
+	minimistOptions = decamelizeKeys(minimistOptions, '-', {exclude: ['stopEarly', '--']});
 
 	if (options.inferType) {
-		delete minimistoptions.arguments;
+		delete minimistOptions.arguments;
 	}
 
-	minimistoptions = buildMinimistOptions(minimistoptions);
+	minimistOptions = buildMinimistOptions(minimistOptions);
 
-	if (minimistoptions['--']) {
-		minimistoptions.configuration = {
-			...minimistoptions.configuration,
+	if (minimistOptions['--']) {
+		minimistOptions.configuration = {
+			...minimistOptions.configuration,
 			'populate--': true
 		};
 	}
 
+	if (minimistOptions.array !== undefined) {
+		minimistOptions.array = arrify(minimistOptions.array).map(flagKey => ({
+			key: flagKey,
+			[options.flags[flagKey].type || 'string']: true
+		}));
+	}
+
```

Comment:
This should preferably be supported by `minimist-options`, but I can't really see how as it uses `type: 'array'` and thus lose the sub-typing (boolean, number etc.).